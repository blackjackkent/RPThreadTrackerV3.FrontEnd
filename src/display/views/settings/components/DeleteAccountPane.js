import React from 'react';
import PropTypes from 'prop-types';
import { TabPane, Button, CardHeader, CardBody } from 'reactstrap';
import Card from '../../../shared/styled/Card';

const propTypes = {
	onDeleteAccountClicked: PropTypes.func.isRequired
};
const DeleteAccountPane = ({ onDeleteAccountClicked }) => {
	return (
		<TabPane tabId="delete-account">
			<Card>
				<CardHeader>
					<i className="fas fa-trash-alt" /> Delete Account
				</CardHeader>
				<CardBody className="card-body text-center">
					<Button
						data-spec="delete-account-button"
						type="submit"
						color="danger"
						onClick={onDeleteAccountClicked}
					>
						Delete Account
					</Button>
				</CardBody>
			</Card>
		</TabPane>
	);
};
DeleteAccountPane.propTypes = propTypes;
export default DeleteAccountPane;
