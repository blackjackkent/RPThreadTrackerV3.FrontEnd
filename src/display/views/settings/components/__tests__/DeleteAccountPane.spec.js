// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '~/testhelpers/helpers.unit';
import DeleteAccountPane from '../DeleteAccountPane';
// #endregion imports

const createTestProps = (propOverrides) => ({
	onDeleteAccountClicked: jest.fn(),
	...propOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const element = shallow(<DeleteAccountPane {...createTestProps()} />);
			expect(element).toMatchSnapshot();
		});
	});
});

describe('behavior', () => {
	describe('onDeleteAccountClicked', () => {
		it('should be called when button is clicked', () => {
			const onDeleteAccountClicked = jest.fn();
			const props = createTestProps({
				onDeleteAccountClicked
			});
			const element = shallow(<DeleteAccountPane {...props} />);
			const btn = getSpecWrapper(element, 'delete-account-button');
			btn.simulate('click');
			expect(onDeleteAccountClicked).toHaveBeenCalledTimes(1);
		});
	});
});
