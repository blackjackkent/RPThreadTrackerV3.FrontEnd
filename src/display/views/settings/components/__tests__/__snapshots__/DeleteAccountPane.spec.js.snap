// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<TabPane
  tabId="delete-account"
  tag="div"
>
  <Styled(Card)>
    <CardHeader
      tag="div"
    >
      <i
        className="fas fa-trash-alt"
      />
       Delete Account
    </CardHeader>
    <CardBody
      className="card-body text-center"
      tag="div"
    >
      <Button
        color="danger"
        data-spec="delete-account-button"
        onClick={[MockFunction]}
        tag="button"
        type="submit"
      >
        Delete Account
      </Button>
    </CardBody>
  </Styled(Card)>
</TabPane>
`;
